### PR TITLE
Ehub 252 formatting options bundle group

### PR DESCRIPTION
**Pull request by ianasqazi**
Hello @nshaw , 

I have installed and tested the functionality of ckeditor with the hub environment. Some details of this change as follows :
- Implemented using the integration with existing forms way - https://ckeditor.com/docs/ckeditor5/latest/installation/advanced/saving-data.html#automatic-integration-with-html-forms
- Installed new library **ckeditor-classic build** (integration of forms is only possible with classic editor build)
- Installed new library  **html-react-parser** which I am using when rending the view data in following 2 files : 
1.   application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/bundle-group/BundleGroupPage.js
2.   application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/CatalogTile.js
- When the pageload event is complete, the useEffect will run and replace the TextArea to CKEditor elements. 

Pending: 
- the toolbar elements are add on plugins to install/imported for the ones that are needed. I have commented in the function replaceTextAreaToCKEditor the toolbar section as samples. 
- Here is a Full feature Editor with all the options in the toolbar (https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/full-featured-editor.html#demo). Can you confirm which ones we need to enable and which we don't need and I will install those modules which are not there.

### CODE REVIEW

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/components/forms/BundleGroupForm/BundleGroupForm.js [hunk 1]
```diff
@@ -314,6 +314,53 @@ const BundleGroupForm = ({
...
+                    toolbar: {
+                        items: [
+                            'undo', 'redo',
+                            '|',
+                            'selectAll',
+                            '|',
+                            'heading',
+                            '|',
+                            'bold', 'italic',
+                            // 'underline', 'code', 'subscript', 'superscript',
```
Comment (nshaw) [inline application/ui/widgets/eh-widgets-dir/eh-widgets/src/components/forms/BundleGroupForm/BundleGroupForm.js:331]: Do you disable these? I tried enabling locally and didn't see a change so I'm not quite sure what to make of it.

**Review by nshaw — CHANGES_REQUESTED** [review]
This looks pretty promising but I noted a couple issues on EHUB-252 which should be addressed.

**Comment by ianasqazi** [comment]
Hello @nshaw ,

I have updated the PR to now include the [REACT-CKEditor build](https://ckeditor.com/docs/ckeditor5/latest/installation/getting-started/frameworks/react.html). Kindly pull and test in local to check the functionality. This is the react classic base edition. 

* I have tested the view/updating of data in the description box. 
* The validations (word cound check) is working too : Shows error in the console but need to create a new element to show the error warnings(which was earlier in the default form error field to highlight red and show error msg). 
* Currently I have used all the header options that come with the base edition for plugins. 
* Commented the <TextArea> code for now. Will remove it if not needed going forward.

**Comment by nshaw** [comment]
@anasliferay I ran some local tests and this isn't working for me. The editor has very few options, some of the ones there don't work yet (lists), and the formatting in the details page is completely unrelated to what you see when editing. I just scheduled a call for tomorrow to review.

**Comment by ianasqazi** [comment]
This pull request was closed due to the limitations with the library (CKEditor) and its integration with our existing html form environment. 

The default build of react-module + classic-build gives us less options to customize the header plugins. If we need to modify the plugins the npm-bundle needs to be updated manually and built with webpack. 

We have tested the deployment of bundled ckeditor but importing the package into our repository would not be a good approach and decided to try other editors with react.

**Pull request closed without merging**